forte/data/base_pack.py

```diff
@@ -97,6 +98,7 @@ def __init__(self, pack_name: Optional[str] = None):
         super().__init__()
         self.links: List[LinkType] = []
         self.groups: List[GroupType] = []
+        self.pack_version: List[int] = PACK_VERSION
```

Comment:
Since a version spec should only contain 3 fields, we shouldn't use `List` but should store these explicitly, probably as `NamedTuple`?